v0/destinations/facebook_offline_conversions/utils.js

```diff
@@ -0,0 +1,379 @@
+    const eventSetIds = getEventSetIds(eventsToIds, standardEvent);
+    if (eventSetIds.length) {
+      payload.push({ standardEvent, eventSetIds });
+    }
+  });
+  if (!payload.length) {
+    throw new ErrorBuilder()
+      .setMessage(
+        "[Facebook Offline Conversions] :: Please Map Your Standard Events With Event Set Ids"
+      )
+      .setStatus(400)
+      .setStatTags({
+        destType: DESTINATION,
+        stage: TRANSFORMER_METRIC.TRANSFORMER_STAGE.TRANSFORM,
+        scope: TRANSFORMER_METRIC.MEASUREMENT_TYPE.TRANSFORMATION.SCOPE,
+        meta:
+          TRANSFORMER_METRIC.MEASUREMENT_TYPE.TRANSFORMATION.META.CONFIGURATION
+      })
+      .build();
+  }
+
+  return payload;
+};
+
+/**
+ * Returns content type
+ * @param {*} message
+ * @returns
+ */
+const getContentType = message => {
+  let contentType = "product";
+  const { event } = message;
+  if (event === "Product List Viewed") {
+    contentType = "product_group";
+  }
+  return contentType;
+};
+
+/**
+ * Returns an array of products
+ * @param {*} contents
+ * @returns
+ */
+const getProducts = contents => {
+  const fbContents = [];
+  const products = [];
+  if (Array.isArray(contents)) {
+    fbContents.push(...contents);
+  } else {
+    fbContents.push(contents);
+  }
+
+  fbContents.forEach(content => {
+    const id = content.product_id || content.id || content.sku;
+    const quantity = content.quantity || 1;
+    const { price, brand, category } = content;
+    if (id) {
+      const obj = { id, quantity, price, brand, category };
+      products.push(removeUndefinedAndNullValues(obj));
+    }
+  });
+  return products;
+};
+
+/**
+ * Returns actionSource value
+ * @param {*} payload
+ * @returns
+ */
+const getActionSource = payload => {
+  let actionSource = null;
+  if (
+    payload.action_source &&
+    ACTION_SOURCES_VALUES.includes(payload.action_source)
+  ) {
+    actionSource = payload.action_source;
+  }
+  return actionSource;
+};
+
+/**
+ * Returns the data array
+ * @param {*} payload
+ * @param {*} message
+ * @returns
+ */
+const prepareData = (payload, message, destination) => {
+  const { Config } = destination;
+  const { limitedDataUSage, valueFieldIdentifier } = Config;
+
+  const data = {
+    match_keys: prepareMatchKeys(payload, message),
+    event_time: payload.event_time,
+    currency: payload.currency || "USD",
+    value: get(message.properties, valueFieldIdentifier) || payload.value || 0,
+    content_type: getContentType(message),
+    order_id: payload.order_id || null,
+    item_number: payload.item_number || null,
+    contents: payload.contents ? getProducts(payload.contents) : null,
+    custom_data: message.properties
+      ? extractCustomFields(
+          message,
+          payload,
+          ["properties"],
+          TRACK_EXCLUSION_FIELDS
+        )
+      : null,
+    action_source: getActionSource(payload),
+    event_source_url: payload.event_source_url || null
+  };
+  if (limitedDataUSage) {
```

Comment:
Lets add this ref: https://developers.facebook.com/docs/marketing-apis/data-processing-options here in comments